implementations/rust/ockam/ockam_app_lib/src/shared_service/relay/create.rs

```diff
@@ -13,29 +14,44 @@ use tracing::{debug, info, trace, warn};
 
 impl AppState {
     /// Try to create a relay until it succeeds.
-    pub async fn create_relay(
-        &self,
-        context: Arc<Context>,
-        cli_state: CliState,
-        node_manager: Arc<InMemoryNode>,
-    ) {
+    pub async fn refresh_relay(&self) {
+        let cli_state = self.state().await;
+        let node_manager = self.node_manager().await;
+        let context = self.context();
+
         if !self.is_enrolled().await.unwrap_or(false) {
+            self.update_orchestrator_status_if(
+                OrchestratorStatus::Disconnected,
+                vec![
+                    OrchestratorStatus::Connected,
+                    OrchestratorStatus::Connecting,
+                ],
+            );
```

Comment:
If we want to switch to `Disconnected`, why not do it unconditionally and remove that function call?

Reply:
if the relay refresh gets called during an enrollment, the enrollment phases would be overwritten by the `Disconnected` status.
Ideally, the connection status and the enrollment status should be separated, it's unified for semplicity.

Reply:
Ok got it!